server/service/testing_utils.go

```diff
@@ -253,12 +253,13 @@ func RunServerForTestsWithDS(t *testing.T, ds fleet.Datastore, opts ...*TestServ
 	}
 	limitStore, _ := memstore.New(0)
 	r := MakeHandler(svc, cfg, logger, limitStore, WithLoginRateLimit(throttled.PerMin(100)))
-	server := httptest.NewServer(r)
+	server := httptest.NewUnstartedServer(r)
+	// Set the same ReadTimeout as the actual server
+	server.Config.ReadTimeout = 25 * time.Second
```

Comment:
Oh so the race was here? So weird it didn't trigger in the PR or my local tests

Reply:
PR's CI checks do not run with `-race`.

Reply:
Hm, then I suppose I ran it wrong locally.

Reply:
It's sporadic though, I managed to reproduce if running ALL tests under `cmd/fleetctl/`. Anyways, good we have race enabled in the daily run. We should check how long it takes, maybe we can add it to PRs too...

Reply:
~48m, forget what I said :)